include/modem/lte_lc.h

```diff
@@ -214,21 +214,27 @@ enum lte_lc_evt_type {
 	 */
 	LTE_LC_EVT_NW_REG_STATUS,
 
+#if defined(CONFIG_LTE_LC_PSM) || defined(CONFIG_ZTEST) || defined(CONFIG_UNITY) || \
```

Comment:
I think you can get rid of `defined(CONFIG_ZTEST) || defined(CONFIG_UNITY)` flagging if you set `CONFIG_LTE_LC_PSM` in the test `CMakeLists.txt` that doesn't enable `CONFIG_LTE_LINK_CONTROL` as follows:
```
target_compile_options(app
  PRIVATE
  -DCONFIG_LTE_LINK_CONTROL_LOG_LEVEL=0
  -DCONFIG_LTE_NEIGHBOR_CELLS_MAX=10
  -DCONFIG_LTE_LC_PSM=y
)
```
lte_lc test set have defined few Kconfig options here earlier and we could extend the list.
I didn't test this fully but something quickly seemed to maybe work 😄

Reply:
I don't remember if you need to actually have `-DCONFIG_LTE_LC_PSM=y`, `-DCONFIG_LTE_LC_PSM=1` or `-DCONFIG_LTE_LC_PSM` but you can try it out.